packages/commonwealth/server/routes/tokenBalance.ts

```diff
@@ -50,12 +59,50 @@ const tokenBalance = async (
   }
 
   try {
-    const balance = await tokenBalanceCache.fetchUserBalance(
-      chain.network,
-      chain_node_id,
-      req.body.address,
-      req.body.contract_address
-    );
+    let balance:
+      | string
+      | {
+          address: string;
+          balance: string;
+        }[];
+    if (req.body.all) {
+      const user_id = await models.Address.findOne({
```

Comment:
We can use the `req.user.id` field to access the requester user id if the route is jwt gated -- probably a better solution than relying on the address field when `all` parameter is passed.

Reply:
Good catch/idea. Added this